daemon/graphdriver/overlay2/mount.go

```diff
@@ -57,7 +58,6 @@ func mountFrom(dir, device, target, mType string, flags uintptr, label string) e
 	if err := json.NewEncoder(w).Encode(options); err != nil {
 		return fmt.Errorf("mountfrom json encode to pipe failed: %v", err)
 	}
-	w.Close()
```

Comment:
The close was explicitly called before `Wait` before to ensure that the process is not waiting on additional input.

Reply:
In that case, should a `w.Close()` be put inside the err ifs?

Reply:
And probably a comment added to prevent someone from changing to a defer

Reply:
@thaJeztah I think it is more a matter of style, I prefer to have `w.Close` inside the if statements but calling `w.Close` multiple times (one before Wait, and one on defer) is not a problem.